### PR TITLE
8344071: Mark some jdk/jfr/event/oldobject test flagless until they fixed to support all GC

### PR DESCRIPTION
**Pull request by lmesnik**
Tests
jdk/jfr/event/oldobject/TestObjectDescription.java
jdk/jfr/event/oldobject/TestClassLoaderLeak.java
fail with different GCs to get expected events.

They pass with G1 GC so I don't want to problemlist them.
However, they should test all GCs that support these events.

Removed ZGC, because I expect that is Generational and should be also supported.

The flagless should be removed when the main bugs are fixed.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8344071](https://bugs.openjdk.org/browse/JDK-8344071): Mark some jdk/jfr/event/oldobject test flagless until they fixed to support all GC (**Bug** - P4)

### Reviewers
 * [Stefan Karlsson](https://openjdk.org/census#stefank) (@stefank - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/22050/head:pull/22050` \
`$ git checkout pull/22050`

Update a local copy of the PR: \
`$ git checkout pull/22050` \
`$ git pull https://git.openjdk.org/jdk.git pull/22050/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 22050`

View PR using the GUI difftool: \
`$ git pr show -t 22050`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/22050.diff">https://git.openjdk.org/jdk/pull/22050.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/22050#issuecomment-2471731897)
</details>

### CODE REVIEW

File: test/jdk/jdk/jfr/event/oldobject/TestObjectDescription.java [hunk 1]
```diff
@@ -41,7 +41,9 @@
  * @test
  * @key jfr
  * @requires vm.hasJFR
- * @requires vm.gc != "Z" & vm.gc != "Shenandoah"
```
Comment (stefank) [inline test/jdk/jdk/jfr/event/oldobject/TestObjectDescription.java:44]: Why did you remove the exclusion of ZGC here and left Shenandoah? Isn't it better to leave the ZGC exclusion here?

**Review by stefank — APPROVED** [review]

File: test/jdk/jdk/jfr/event/oldobject/TestObjectDescription.java (hunk 1 above)
Comment (lmesnik) [reply test/jdk/jdk/jfr/event/oldobject/TestObjectDescription.java:44]: The ZGC is now generational, and the OldObjectSampler should work.

Comment (stefank) [reply test/jdk/jdk/jfr/event/oldobject/TestObjectDescription.java:44]: I don't know what the generational part has to do with it. The Old part in the name doesn't refer the young and old generation, just that an object is lingering in the system, which it also did for single gen ZGC.

Comment (lmesnik) [reply test/jdk/jdk/jfr/event/oldobject/TestObjectDescription.java:44]: Thanks, I copied your comments to the bug. It might be needed to run specific tests for Parallel/Serial GC like for ZGC so this will be just G1 only.
